src/convertToObject.js

```diff
@@ -6,7 +6,18 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const stylesObject = sourceString
+    .split(';')
+    .map((styles) => styles.split(':'))
+    .reduce((object, [key, value]) => {
+      if (key && value !== undefined && value.trim() !== '') {
```

Comment:
Insufficient checks before assigning values: The code simply checks key, but does not check that key is not an empty string after removing spaces. This can lead to incorrect keys being added to the object.